app/views/posts/new.html.erb

```diff
@@ -0,0 +1,9 @@
+<%= form_with model: @post, url: posts_path do |form| %>
+  <%= form.label :title %>
+  <%= form.text_field :title %>
+
+  <%= form.label :text %>
+  <%= form.text_area :text %>
+
+  <%= form.submit "Create Post" %>
+<% end %>
```

Comment:
- [ ] Good Job for Implemented   form in the views/posts/new.html.erb file which accepts input for creating posts.But based on your routes for creating posts   `new_user_post  GET    /users/:user_id/posts/new(.:format)` when i checked it I got the following error 
![image](https://github.com/EvansSnave/Blog_app_rails/assets/93598189/82813b1e-1de6-4474-a011-fef5c87a8878)

- This basically because Based on your routes configuration, it appears that `posts_path` is nested under **users.** Therefore, you should use the nested route helper for the form in your `new.html.erb` file. check your routes configuration below.
```
Rails.application.routes.draw do
  root 'users#index'
  
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create, :destroy]
    end
  end
end
```
So kindly  your form_ with tag to include both @user and @post in the model, and use the nested route helper for the URL: just like below
```
<%= form_with model: [@user, @post], url: user_posts_path(@user) do |form| %>
  <%= form.label :title %>
  <%= form.text_field :title %>

  <%= form.label :text %>
  <%= form.text_area :text %>

  <%= form.submit "Create Post" %>
<% end %>
 
```
Then Ensure that you are setting **@user** in the controller action, possibly in the new action of your **PostsController**. For example:
```
def new
  @user = User.find(params[:user_id])
  @post = Post.new
end

```
Make sure that` params[:user_id]` is correctly obtaining the user ID from the route. Then you will get the correct url to create post just like the below screenshot
 
![image](https://github.com/EvansSnave/Blog_app_rails/assets/93598189/84f0e360-5b73-40fa-ba8d-f64cecef3684)

- Here is the first comment I create by following the above suggestion 
![image](https://github.com/EvansSnave/Blog_app_rails/assets/93598189/e9584d98-cc0e-404c-827a-a025ad8c48cf)